### DISCUSSION TITLE
queries should automatically factor in the current filetype

### DISCUSSION THREAD

**Discussion by chrisgrieser** [jackMort/ChatGPT.nvim#18] [Ideas]
Take this example, which takes place while my current file is a lua file. The first query gives me an answer, which is not valid syntax in lua. I to have  explicitly add "in lua" to get the correct answer. Given that the current file is a lua file, I assume it would be smarter if the plugin automatically adds "in {current filetype}" to any query (or offer this as an option).

This would reduce the number of wrong answers, and also reduce the number of queries one has to send when remembering that the filetype is missing (= less tokens)

<img width="811" alt="image" src="https://user-images.githubusercontent.com/73286100/207995159-d88d6154-1688-418a-8d71-08817a1f9fd1.png">

**Comment by jackMort** [comment]
I've been thinking about something similar, I'll try to add options to configure prefixes/suffixes. But it needs to be done more generically, as this plugin may be used in various ways, not just for coding. 

I'm also working on other commands specifically for coding: code completion/refactoring/adding comments, etc. I'll be back soon with the first version.

By the way, what is your custom configuration and what font and colorscheme are you using? It looks pretty good.

**Comment by chrisgrieser** [comment]
great, I think simply having different options would be useful here. Maybe apply the prefix/suffix only for when you have have the ChatUI window open, and not for the other scenarios?

> I'm also working on other commands specifically for coding: code completion/refactoring/adding comments, etc.

great to hear 🥳
for completions, though, I feel like it needs a very different approach, maybe requiring even a different plugin? because for completion, one would want to have integration with plugins like cmp, and also somehow a method to restrict the number of queries (since queries on every keystroke is gonna use a lot of tokens...)

> By the way, what is your custom configuration and what font and colorscheme are you using? It looks pretty good.

Thanks! The font is JetBrains Mono, the theme is [oxocarbon](https://github.com/nyoom-engineering/oxocarbon.nvim)

**Comment by ben-wall** [comment]
Yeah would be good to have an option to launch it like this like `:ChatGPTFiletype`
